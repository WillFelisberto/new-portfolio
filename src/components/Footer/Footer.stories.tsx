import { Meta, StoryObj } from '@storybook/react';
import Footer from '.';

const meta: Meta<typeof Footer> = {
  title: 'Components/Footer',

  component: Footer,
} satisfies Meta<typeof Footer>;

export default meta;
type Story = StoryObj<typeof Footer>;

export const Default: Story = {
  args: {
    //ðŸ‘‡ The args you need here will depend on your component
  },
};
