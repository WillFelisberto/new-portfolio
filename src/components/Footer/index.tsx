import { IoMdHeart } from 'react-icons/io';

type FooterProps = {
  textFooterMade: string;
  textFooterBy: string;
};

const Footer = ({ textFooterMade, textFooterBy }: FooterProps) => {
  return (
    <footer className="flex h-14 w-full items-center justify-center bg-gray-950">
      <span className="flex items-center gap-1.5 font-mono text-xs text-gray-400 sm:text-sm">
        {textFooterMade}
        <IoMdHeart aria-label="Heart icon" size={13} className="text-sky-500" />
        {textFooterBy}
        <strong className="font-medium">Willian Souza</strong>
      </span>
    </footer>
  );
};

export default Footer;
