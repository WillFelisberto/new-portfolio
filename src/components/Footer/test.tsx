import { render, screen } from '@testing-library/react';

import Footer from '.';

describe('<Footer />', () => {
  it('should render footer with provided text props', () => {
    const textFooterMade = 'Made with';
    const textFooterBy = 'by';

    const { container } = render(
      <Footer textFooterMade={textFooterMade} textFooterBy={textFooterBy} />,
    );

    expect(screen.getByText(/Made with/i)).toBeInTheDocument();
    expect(screen.getByLabelText(/Heart icon/i)).toBeInTheDocument();
    expect(screen.getByText(/by/i)).toBeInTheDocument();
    expect(screen.getByText(/Willian Souza/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render footer when textFooterMade is empty', () => {
    const textFooterMade = '';
    const textFooterBy = 'by';

    render(
      <Footer textFooterMade={textFooterMade} textFooterBy={textFooterBy} />,
    );

    expect(screen.getByLabelText(/Heart icon/i)).toBeInTheDocument();
    expect(screen.getByText(/by/i)).toBeInTheDocument();
    expect(screen.getByText(/Willian Souza/i)).toBeInTheDocument();
  });
});
