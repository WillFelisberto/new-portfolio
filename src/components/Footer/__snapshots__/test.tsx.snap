// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render footer with provided text props 1`] = `
<footer
  class="flex h-14 w-full items-center justify-center bg-gray-950"
>
  <span
    class="flex items-center gap-1.5 font-mono text-xs text-gray-400 sm:text-sm"
  >
    Made with
    <svg
      aria-label="Heart icon"
      class="text-sky-500"
      fill="currentColor"
      height="13"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="13"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 448l-30.164-27.211C118.718 322.442 48 258.61 48 179.095 48 114.221 97.918 64 162.4 64c36.399 0 70.717 16.742 93.6 43.947C278.882 80.742 313.199 64 349.6 64 414.082 64 464 114.221 464 179.095c0 79.516-70.719 143.348-177.836 241.694L256 448z"
      />
    </svg>
    by
    <strong
      class="font-medium"
    >
      Willian Souza
    </strong>
  </span>
</footer>
`;
